features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/Listing/Edit.jsx

```diff
@@ -0,0 +1,305 @@
+        }).catch(() => {
+            Alert.error(intl.formatMessage({
+                defaultMessage: 'Error while retrieving service and schema types',
+                id: 'ServiceCatalog.Listing.Edit.error.retrieve.service.schema.types',
+            }));
+        });
+    }, []);
+
+    const handleChange = (e) => {
+        dispatch({ field: e.target.name, value: e.target.value });
+    };
+
+    /**
+     * Function for updating a given service entry
+     */
+    function doneEditing() {
+        const updateServicePromise = ServiceCatalog.updateService(id, state);
+        updateServicePromise.then(() => {
+            Alert.info(intl.formatMessage({
+                id: 'ServiceCatalog.Listing.Delete.service.updated.successfully',
+                defaultMessage: 'Service updated successfully!',
+            }));
+            // Reload the services list
+            getData();
+        }).catch(() => {
+            Alert.error(intl.formatMessage({
+                defaultMessage: 'Error while updating service',
+                id: 'ServiceCatalog.Listing.Edit.error.update',
+            }));
+        });
+        setOpen(!open);
+    }
+
+    return (
+        <>
+            <Button onClick={toggleOpen}>
+                <Icon>edit</Icon>
+            </Button>
+            <Dialog
+                open={open}
+                disableBackdropClick
+                disableEscapeKeyDown
+                maxWidth='sm'
+                fullWidth
+                aria-labelledby='confirmation-dialog-title'
+            >
+                <DialogTitle id='confirmation-dialog-title'>
+                    <FormattedMessage
+                        id='ServiceCatalog.Listing.Edit.service'
+                        defaultMessage='Edit Service'
+                    />
+                </DialogTitle>
+                <DialogContent>
+                    <TextField
+                        name='name'
+                        label={(
+                            <>
+                                <FormattedMessage
+                                    id='ServiceCatalog.Listing.Edit.service.name.label'
+                                    defaultMessage='Service'
+                                />
+                                <sup className={classes.mandatoryStar}>*</sup>
+                            </>
+                        )}
+                        value={name}
+                        variant='outlined'
+                        fullWidth
+                        helperText={(
+                            <FormattedMessage
+                                id='ServiceCatalog.Listing.Edit.service.name.helper.text'
+                                defaultMessage='Name of the service'
+                            />
+                        )}
+                        onChange={handleChange}
+                    />
+                </DialogContent>
+                <DialogContent>
```

Comment:
I think having multiple `<DialogContent>`s inside a dialog is wrong, If you need to add a padding(spacing) between two elements use `Grid` spacing

Reply:
[MUI Doc](https://material-ui.com/components/dialogs/#dialog)

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9407/commits/07b6f26cf83d2651ab4ce4e82968a338e75952cd